### DISCUSSION TITLE
Questions with Postgres tables

### DISCUSSION THREAD

**Discussion by GuilhermeHirsch** [Budibase/budibase#1927] [Help] [answered]
I'm trying to reference a Postgres table in "Data Provider". However, "No rows found" is returned. Is it only possible to work with queries when it comes to Postgres tables? 
Another question that arose: How do I return the "current date" in an action (update) of a table in automation?

**Comment by joebudi** [comment]
Hey @GuilhermeHirsch 

In re to your data provider/PostgreSQL issue, did you follow the steps below:

- [ ] Add PostgreSQL as a data provider
- [ ] Complete config
- [ ] Click the fetch data from database button

Did tables from your PostgreSQL database appear? Was there data in the tables?

I am not exactly sure what you want to do in re to the "current date" - could you elaborate, please?

**Reply by GuilhermeHirsch** [reply]
Yes, the PostgreSQL database appears in the "data" tab. I configured the "data provider" and redirected it in the "table". With the query I created for testing it works.

![a](https://user-images.githubusercontent.com/33108596/124587606-2e34b480-de2e-11eb-944b-a7a3efdb05b5.jpg)

As for the date, I want to create a button in a table that when clicked updates the "Execution Date" field of the row with the current date. I thought that this was possible through the activation of an automation.

**Reply by joebudi** [reply]
This is a strange one. @mike12345567 could you help with this?

**Comment by GuilhermeHirsch** [comment, marked as answer]
I made a video of the first question to help: https://www.loom.com/share/c85dddbf4ddc4930b05dc4c6f003adae

**Reply by mike12345567** [reply]
Hi @GuilhermeHirsch - it looks like there is some sort of issue with the primary key constraint processing going on, not 100% sure whats happening here as would need to see the underlying schema of the SQL table, but noticed this in the video:
![image](https://user-images.githubusercontent.com/4407001/124606809-25aba080-de65-11eb-974b-355999a736b2.png)

Would you be able to give me any more info about how your table keys are defined? Realistically this shouldn't have happened and I'll look into a fix for when it detects many primary keys with the same name, but some idea of the table definition might help.

**Reply by GuilhermeHirsch** [reply]
I deleted the database and generated it again, but it continues with the same behavior of "ids" that you mentioned.
The complete imported table is this:
![Sem título](https://user-images.githubusercontent.com/33108596/124609191-cb4f1780-de45-11eb-9680-08b205129678.jpg)

**Reply by mike12345567** [reply]
Interesting, I'll need to look into this, one thing I would say as well, I looked into the builder and I think if you add a data provider, immediately add a table and then setup the data provider the table will not be fully configured until you select it again.

Can you try clicking on the table which is display "no rows found" in the builder, and seeing if it updates correctly after selecting it?

Clicking this might fix it:
![image](https://user-images.githubusercontent.com/4407001/124609897-f64a6300-de67-11eb-9eb0-fc24cd204398.png)

**Reply by GuilhermeHirsch** [reply]
Same behavior: https://www.loom.com/share/6895c7f5eed34c7b8dbdd68853e1bfaf

**Reply by shogunpurple** [reply]
@GuilhermeHirsch If possible, it would be super helpful to see the `PRIMARY KEY` constraint on your table. I'm thinking that you have a composite key on `id` and `version`.

Does your definition look something like this?

```
 PRIMARY KEY(id, version)
```

The system does not support composite keys at the moment - there's an issue for this here:
https://github.com/Budibase/budibase/issues/1738

**Reply by GuilhermeHirsch** [reply]
I created my database in Supabase. The id type is uuid (I don't know if it makes any difference in this case).

**Reply by shogunpurple** [reply]
Ah, that seems odd. We don't account directly for UUID, but it should default to string. My question was what your primary key is - you should be able to see this in the schema configuration of your supabase table

**Reply by GuilhermeHirsch** [reply]
I questioned this fact on the Supabase forum.

**Reply by GuilhermeHirsch** [reply]
Thinking and researching better, I will try to use the internal tables. I think it's great for my needs, it's something I discover over time. Thanks for your help!

**Reply by GuilhermeHirsch** [reply]
> I questioned this fact on the Supabase forum.

Not to leave this topic without feedback. I was informed that this does not happen automatically. After that, I created a new database and integrated it with Budibase. Surprisingly everything worked out, and the tables appeared.

**Comment by GuilhermeHirsch** [comment]
Thinking and researching better, I will try to use the internal tables. I think it's great for my needs, it's something I discover over time. Thanks for your help!
